internal/p2p/address.go

```diff
@@ -97,7 +97,7 @@ func ParseNodeAddress(urlString string) (NodeAddress, error) {
 
 // Resolve resolves a NodeAddress into a set of Endpoints, by expanding
 // out a DNS hostname to IP addresses.
-func (a NodeAddress) Resolve(ctx context.Context) ([]Endpoint, error) {
+func (a NodeAddress) Resolve(ctx context.Context) ([]*Endpoint, error) {
```

Comment:
```suggestion
func (a NodeAddress) Resolve(ctx context.Context) (*Endpoint, error) {
```

(I think you meant?)

Reply:
I think this one should stay the way that it is because the DNS resolver can return multiple endpoints (say we get a hostname with multiple A records, etc.) then this ends up being the correct thing to do?)